### PR TITLE
Use createdAt as head of imported account

### PR DESCRIPTION
**Pull request by hairtail**
## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

**Comment by dguenther** [comment]
Let me take a look at this -- I was just running into something similar while testing imports on the Node App

File: ironfish/src/wallet/wallet.ts [hunk 1]
```diff
@@ -1544,7 +1544,7 @@ export class Wallet {
 
     await this.walletDb.db.transaction(async (tx) => {
       await this.walletDb.setAccount(account, tx)
-      await account.updateHead(null, tx)
+      await account.updateHead(createdAt, tx)
```
Comment (hughy) [inline ironfish/src/wallet/wallet.ts:1547]: I think this is a good idea, but it will take some work to go through all of the existing tests and patch them for the new behavior and make sure that this doesn't introduce any regressions in edge cases

Comment (hughy) [reply ironfish/src/wallet/wallet.ts:1547]: I spent some time debugging the test failures, and I think the issue is that we need to use the block _before_ `createdAt` here so that the the `createdAt` block is still scanned for the account.

For example, if an account receives tokens in the genesis block then it has a `createdAt` that refers to the genesis block. It will need to scan the genesis block in order to have a correct balance.

Comment (hairtail) [reply ironfish/src/wallet/wallet.ts]: Thanks for fixing tests!

**Review by dguenther — APPROVED** [review]
Tested this out a bit and doesn't seem like there are any functionality regressions, thanks for the improvement!
